sklearn/preprocessing/_encoders.py

```diff
@@ -572,11 +671,30 @@ def _transform_new(self, X):
 
         X_int, X_mask = self._transform(X, handle_unknown=self.handle_unknown)
 
+        if self.drop is not None:
+            for i in range(n_features):
+                Xii = X_int[:, i]
+                Xmi = X_mask[:, i]
+                drop_value = self.drop_idx_[i]
+                """
+                Add the cells where the drop value is present to the list
+                of those to be masked-out. Decrement the indices of the values
```

Comment:
I see three options for implementing `transform`:
1. ignore the added category when encoding, a bit like handle_unknown='ignore'
2. remove the value once encoded as an int
3. remove the columns once one-hot encoded

I think you've implemented #2. Why is this one chosen? What are its pros and cons in terms of code simplicity and computational complexity?

Reply:
To do the first option, it seems like I'd have to modify _transform(), which is in the BaseEncoder class. We're not adding drop options to the other classes that inherit from BaseEncoder, so I thought that changing this in BaseEncoder might not be the best approach. 
The approach I took with 2) had some advantages, since I was able to mix it in to reuse some of the logic that was already used for masking unknown values. It also avoids having to do column slicing on an output that might be a CSR matrix (as would be the case in 3)).

Reply:
@jnothman do you think that one of the other approaches is superior?